src/stellar-wallet.tsx

```diff
@@ -85,17 +86,24 @@ const fetchAllPositions = async (user: string): Promise<PositionsRecord> => {
   return Object.fromEntries(positionsArr);
 };
 
-const createBalanceRecord = (balances: Balance[]): BalanceRecord =>
-  balances.reduce((acc, balance) => {
-    if (balance.asset_type === 'native') {
-      acc.XLM = balance;
-    } else if (balance.asset_type === 'credit_alphanum4' && balance.asset_code === 'USDC') {
-      acc.USDC = balance;
-    } else if (balance.asset_type === 'credit_alphanum4' && balance.asset_code === 'EURC') {
-      acc.EURC = balance;
-    }
-    return acc;
-  }, {} as BalanceRecord);
+const createBalanceRecord = (balances: StellarSdk.Horizon.HorizonApi.BalanceLine[]): BalanceRecord =>
+  balances.reduce(
+    (acc, balanceLine) => {
+      if (balanceLine.asset_type === 'native') {
+        acc.XLM = { trustLine: true, balanceLine };
+      } else if (balanceLine.asset_type === 'credit_alphanum4' && isSupportedCurrency(balanceLine.asset_code)) {
+        acc[balanceLine.asset_code] = { trustLine: true, balanceLine };
+      }
+      return acc;
+    },
+    {
+      XLM: { trustLine: false },
```

Comment:
I might misunderstand it, but why is it trustline: false for XLM?

Reply:
Yeah, it could default to true, but that kind of breaks the typing, as we would have to give the BalanceLine object as well. The trustLine will get set to true when the balances load from Horizon, so I think this can stay as is, being the simplest solution. We can consider the trustLine to kind of not exist if there's no balanceLine, make sense?

I agree it's kinda logically incorrect tho.